diff -uraN vapoursynth-R56/Makefile.am omv-vapoursynth-R56/Makefile.am
--- vapoursynth-R56/Makefile.am	2021-09-29 13:22:01.000000000 +0200
+++ omv-vapoursynth-R56/Makefile.am	2021-10-06 11:58:58.007249189 +0200
@@ -88,7 +88,7 @@
 
 pkgconfig_DATA += pc/vapoursynth.pc
 
-libvapoursynth_la_LDFLAGS = -no-undefined -avoid-version $(UNDEFINEDLDFLAGS)
+libvapoursynth_la_LDFLAGS = -no-undefined -release $(PACKAGE_VERSION)
 libvapoursynth_la_CPPFLAGS = $(PTHREAD_CFLAGS) $(ZIMG_CFLAGS) -DVS_PATH_PLUGINDIR='"$(PLUGINDIR)"'
 libvapoursynth_la_LIBADD = $(PTHREAD_LIBS) $(ZIMG_LIBS) $(DLOPENLIB)
 
